src/Estimators/src/SubModel.cpp

```diff
@@ -0,0 +1,409 @@
+        return false;
+    }
+
+    auto populateSensorParameters = [&ptr, logPrefix](const std::string& groupName, std::vector<std::string>& names, std::vector<std::string>& frames) -> bool {
+        auto group = ptr->getGroup(groupName).lock();
+        if (group == nullptr)
+        {
+            blf::log()->error("{} Unable to get the group named '{}'.", logPrefix, groupName);
+            return false;
+        }
+
+        if (!group->getParameter("names", names))
+        {
+            blf::log()->error("{} The parameter handler could not find 'names' in the configuration file for the group {}.", logPrefix, groupName);
+            return false;
+        }
+
+        if (!group->getParameter("frames", frames))
+        {
+            blf::log()->error("{} The parameter handler could not find 'frames' in the configuration file for the group {}.", logPrefix, groupName);
+            return false;
+        }
+        return true;
+    };
+
+    auto extContactGroup = ptr->getGroup("EXTERNAL_CONTACT").lock();
+    if (extContactGroup == nullptr)
+    {
+        blf::log()->error("{} Unable to get the group named 'EXTERNAL_CONTACT'.", logPrefix);
+        return false;
+    }
+
+    std::vector<std::string> extContactFrames;
+    if (!extContactGroup->getParameter("frames", extContactFrames))
+    {
+        blf::log()->warn("{} The parameter handler could not find \" frames \" in the configuration file.", logPrefix);
+    }
+
+    std::vector<std::string> ftNames, ftFrames;
+    bool ok = populateSensorParameters("FT", ftNames, ftFrames);
+
+    std::vector<blfEstim::Sensor> ftList;
+    for (auto idx = 0; idx < ftNames.size(); idx++)
+    {
+        blfEstim::Sensor ft;
+        ft.name = ftNames[idx];
+        ft.frame = ftFrames[idx];
+        ftList.push_back(std::move(ft));
+    }
+
+    std::vector<std::string> accNames, accFrames;
+    ok = ok && populateSensorParameters("ACCELEROMETER", accNames, accFrames);
+
+    std::vector<blfEstim::Sensor> accList;
+    for (auto idx = 0; idx < accNames.size(); idx++)
+    {
+        blfEstim::Sensor acc;
+        acc.name = accNames[idx];
+        acc.frame = accFrames[idx];
+        accList.push_back(std::move(acc));
+    }
+
+    std::vector<std::string> gyroNames, gyroFrames;
+    ok = ok && populateSensorParameters("GYROSCOPE", gyroNames, gyroFrames);
+
+    std::vector<blfEstim::Sensor> gyroList;
+    for (auto idx = 0; idx < gyroNames.size(); idx++)
+    {
+        blfEstim::Sensor gyro;
+        gyro.name = gyroNames[idx];
+        gyro.frame = gyroFrames[idx];
+        gyroList.push_back(std::move(gyro));
+    }
+
+    ok = ok && blfEstim::SubModelCreator::createSubModels(ftList,
+                                      accList,
+                                      gyroList,
+                                      extContactFrames);
+
+    return ok;
+}
+
+void blfEstim::SubModelCreator::setModel(const iDynTree::Model & model)
+{
+    this->model = model.copy();
```

Comment:
Here in principle you can simply use 
```
this->model = model;
```
because in C++ the copy operator is called https://github.com/robotology/idyntree/blob/988c915f82a485fb44925a5a1e80543db75fcf7a/src/model/src/Model.cpp#L81-L90

Right now you are making an extra copy since the ``model.copy()`` generates a temporary model that is then copied in ``this->model`` via the same copy operator I mentioned above, and then destroyed immediately afterward.

Also, to avoid confusion to a reader, it is often preferable to have some naming convention for the private members, like ``m_model``. In this way, 
```cpp
m_model = model;
```
clearly copies an input object to a private class member.